webui/react/src/pages/Cluster/ClusterHistoricalUsage.tsx

```diff
@@ -0,0 +1,218 @@
+import { Button, Col, Row } from 'antd';
+import dayjs from 'dayjs';
+import React, { useEffect, useState } from 'react';
+
+import Section from 'components/Section';
+import { parseUrl } from 'routes/utils';
+import { getResourceAllocationAggregated } from 'services/api';
+
+import css from './ClusterHistoricalUsage.module.scss';
+import ClusterHistoricalUsageChart from './ClusterHistoricalUsageChart';
+import ClusterHistoricalUsageCsvModal from './ClusterHistoricalUsageCsvModal';
+import ClusterHistoricalUsageFilters, {
+  ClusterHistoricalUsageFiltersInterface,
+} from './ClusterHistoricalUsageFilters';
+import { mapResourceAllocationApiToChartSeries, ResourceAllocationChartSeries } from './utils';
+
+export const DEFAULT_RANGE_DAY = 14;
+export const DEFAULT_RANGE_MONTH = 6;
+export const MAX_RANGE_DAY = 31;
+export const MAX_RANGE_MONTH = 36;
+
+export enum GroupBy {
+  Day = 'day',
+  Month = 'month',
+}
+
+const ClusterHistoricalUsage: React.FC = () => {
+  const [ chartSeries, setChartSeries ] = useState<ResourceAllocationChartSeries|null>(null);
```

Comment:
**nit:** use the `undefined` behavior?
```
const [ chartSeries, setChartSeries ] = useState<ResourceAllocationChartSeries>();
```
Then it should naturally be either `ResourceAllocationChartSeries` or `undefined`?